### PR TITLE
Fix merge dialog on protected branch with missing required statuses

### LINKED ISSUE #10636: bug: protected branch: pull check
Gitea version: 1.12.0+dev-447-g123d38605

if you enable branch protection and leave all default beside "Enable Status Check" the pull merge-status is considderted as not mergable! (only force mergable via admin rights)

to reproduce:
A: enable branch protection
B: enable Status Check **with: no status checks found in the last week ..."

result:
* user with write access:
Bildschirmfoto zu 2020-03-06 13-57-23

* user with admin access:
Bildschirmfoto zu 2020-03-06 13-57-55

### PR DESCRIPTION
**Pull request by 6543**
It is possible for misconfigured protected branches to have required status checks that are not in any of the current statuses: Pending, Success, Error, Failure, or Warning - presumably because the CI has not contacted us as yet.

Fix #10636 by adding case: missing StatusChecks when these are missing

Screenshot at 2020-04-15 13-21-35

### CODE REVIEW

File: templates/repo/issue/view_content/pull.tmpl [hunk 1]
```diff
@@ -58,6 +58,7 @@
 		</div>
 	</div>
 {{end}}
+VARSTATUS: RequiredStatusCheckState.IsSuccess: {{if .RequiredStatusCheckState.IsSuccess}}true{{else}}false{{end}}
```
Comment (6543) [inline templates/repo/issue/view_content/pull.tmpl]:
```suggestion
```

File: templates/repo/issue/view_content/pull.tmpl [hunk 2]
```diff
@@ -66,14 +67,13 @@
...
 	{{- else if .IsBlockedByRejection}}red
-	{{- else if and .EnableStatusCheck (or .RequiredStatusCheckState.IsFailure .RequiredStatusCheckState.IsError)}}red
-	{{- else if and .EnableStatusCheck (or .RequiredStatusCheckState.IsPending .RequiredStatusCheckState.IsWarning)}}yellow
+	{{- else if and .EnableStatusCheck (or (not $.LatestCommitStatus) .RequiredStatusCheckState.IsFailure .RequiredStatusCheckState.IsError)}}red
+	{{- else if and .EnableStatusCheck $.LatestCommitStatus (or .RequiredStatusCheckState.IsPending .RequiredStatusCheckState.IsWarning)}}yellow
 	{{- else if and .RequireSigned (not .WillSign)}}}red
 	{{- else if .Issue.PullRequest.IsChecking}}yellow
 	{{- else if .Issue.PullRequest.CanAutoMerge}}green
 	{{- else}}red{{end}}">{{svg "octicon-git-merge" 32}}</a>
 	<div class="content">
```
Comment (6543) [inline templates/repo/issue/view_content/pull.tmpl]:
```suggestion
	<div class="content">
		{{template "repo/pulls/status" .}}
```

**Review by 6543 — COMMENTED** [review]
rm debugg stuff

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Comment by 6543** [comment]
Backport send  #11084

**Pull request merged**